--- src/core/ngx_conf_file.h	2014-09-15 11:01:46.000000000 +0800
+++ src/core/ngx_conf_file.init_master.h	2014-09-15 10:56:03.000000000 +0800
@@ -114,7 +114,7 @@
     ngx_command_t        *commands;
     ngx_uint_t            type;
 
-    ngx_int_t           (*init_master)(ngx_log_t *log);
+    ngx_int_t           (*init_master)(ngx_cycle_t *cycle);
 
     ngx_int_t           (*init_module)(ngx_cycle_t *cycle);
 
